Jenkinsfile

```diff
@@ -370,8 +370,13 @@ def testGo(prefix) {
         println "Running tests on commit ${env.COMMIT_HASH} with ${goversion}."
         shell "go get \"github.com/stretchr/testify/require\""
         shell "go get \"github.com/stretchr/testify/assert\""
+        def parallelTests = []
         def tests = [:]
         for (def i=0; i<dirs.size(); i++) {
+            if (tests.size() == 4) {
```

Comment:
I think it makes sense to keep `engine` and `chat` in different threads, otherwise, we're not going to see much speedup over fully serial